### PR TITLE
Default response content type using GraphQL spec

### PR DESCRIPTION
**Pull request by ivanp**
So the default content type header was not recognized properly by [GraphQL gateway](https://github.com/apollographql/federation/blob/%40apollo/gateway%402.5.4/gateway-js/src/datasources/RemoteGraphQLDataSource.ts#L299). I tried to [submit PR](https://github.com/apollographql/federation/pull/2769) but they said it doesn't conform to the [GraphQL spec](https://github.com/graphql/graphql-over-http/blob/main/spec/GraphQLOverHTTP.md#serialization-format).

### CODE REVIEW

**Comment by phillip-kruger** [comment]
@jmartisk can you have look ?

**Review by jmartisk — CHANGES_REQUESTED** [review]
Tests are failing - you need to also update https://github.com/quarkusio/quarkus/blob/3.4.0.CR1/extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java#L299 - please update it to accept both the new (fixed) content-type and the original  one, for backward compatibility

**Comment by gsmet** [comment]
@jmartisk this hasn't made any progress and I think it's really something that needs fixing. Could you have a look? Thanks.

**Comment by jmartisk** [comment]
I've added a commit that hopefully completes the fix

File: extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java [hunk 1]
```diff
@@ -297,7 +297,8 @@ private static String getCharset(String mimeType) {
     private boolean isValidAcceptRequest(String mimeType) {
         // At this point we only accept two
```
Comment (gsmet) [inline extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java:298]: Change looks good to fix the tests but this comment looks outdated. I wonder if we should explain a bit more and put the two valid ones first and the last one with a comment `// deprecated content-type supported for compatibility`.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java (hunk 1 above)
Comment (jmartisk) [reply extensions/smallrye-graphql/runtime/src/main/java/io/quarkus/smallrye/graphql/runtime/SmallRyeGraphQLExecutionHandler.java]: Ah, that's a very good observation, thanks. Fixed

**Review by jmartisk — APPROVED** [review]

**Review by phillip-kruger — APPROVED** [review]

**Comment by jmartisk** [comment]
@gsmet can you please have another look? I would merge it but I can't as long as there's a -1 from you :)

**Review by gsmet — APPROVED** [review]
Yeah sorry, I was busy on other things, let's merge!

**Pull request merged**